collector/ims_azure_linux.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	azureComputeURL = "http://169.254.169.254/metadata/instance/compute?api-version=2021-01-01&format=json"
```

Comment:
This reaches out over the network, I'm not sure we want collectors that do this inside the exporter.

Maybe this would be more appropriate as a [textfile collector tool](https://github.com/prometheus-community/node-exporter-textfile-collector-scripts)

Reply:
Yes, but the issue is about the behavior outside of cloud providers. It's an unsecured http connection reaching out from the exporter.

The node_exporter is used in a lot of places that are not cloud.

Reply:
This doesn't change the fact that no collector is connecting anywhere today, so this needs more discussion